src/cbc_sdk/enterprise_edr/threat_intelligence.py

```diff
@@ -509,18 +1139,28 @@ def save_watchlist(self):
         return self
 
     def validate(self):
-        """Validates this report's state.
+        """
+        Validates this report's state.
 
         Raises:
-            InvalidObjectError: If the report's state is invalid
+            InvalidObjectError: If the report's state is invalid.
         """
+        id_substitute = False
```

Comment:
What is the purpose to `id_substitute`?

Reply:
`id_substitute` records whether we had to add a temporary id value before calling the superclass validate, and thus whether we need to _remove_ said temporary id value afterwards. Without this, calling the superclass validate will fail because there's no id in there until it's actually saved.

Reply:
Looking into this I am a little confused why we are doing temporary substitution if these are required properties? If they are not required then we should remove them from the schema file here https://github.com/carbonblack/carbon-black-cloud-sdk-python/blob/develop/src/cbc_sdk/enterprise_edr/models/feed.yaml

A personal preference to reduce the code would be this alternative if the substitution is still needed then you don't need another variable since the string we set is a known no valid one. If this is needed for multiple properties then the cleanup can be a simple for loop through the keys though we should clean up the validation logic before adding substitutes or default to empty strings if a non None is needed

```
if not self._info.get('id', None):
    self._info['id'] = '**TEMP**'
super(Feed, self).validate()
if self._info['id'] === '**TEMP**':
    del self._info['id']
```

Reply:
OK, this is the situation. The `id` value is _technically_ required, but only in `Reports` that have already been created on the server side.  A newly-built `Report` that has not yet saved to the server will not have an `id` yet, as the server assigns it.  However, both `update()` and `save_watchlist()` _call_ `validate()` (which is even more important now as it performs the rebuild of the `iocs_v2` element of `_info` as well), leading us to a chicken-and-egg problem. The `id` substitution lets us get around that problem and lets `validate()` complete OK so the `Report` can be saved.

Reply:
So we can remove the other substitutes and just leave it with the id one?

Reply:
Oh, you're not just looking at `Report`, are you? The fact that this was attached to `Report.validate()` was a bit misleading.

`Feed` requires `owner` and `access` as well. Those are set by the server, and are not required at create time.  So we have to fake those out in addition to `id`.

`Watchlist` requires `create_timestamp` and `last_update_timestamp`, which are also set by the server. At least there I _can_ flag the replacements with a "known not valid" value, namely -1.

Reply:
That doesn't sound like requires as they are not required for creation they are created and managed by the backend which therefore makes them populated and always returned. If they are not required in the request then we don't need to validate them quite as strictly

Reply:
I took the required flags out of the YAML and took the substitutions out of the code. The result passes the unit tests. Does that sit better with you?

Reply:
As long as it functionally works against the CBC with those minimum fields. The required logic should match exactly what is required by the CBC backend. Have you run a manual test against the CBC backend?

Reply:
No, I have not.

Reply:
I've proved out creating Report and Feed without those "missing" fields above.  See comment thread above.

Reply:
And Watchlist creation works, too. Transcript follows.

```
>>> from cbc_sdk import CBCloudAPI
>>> api = CBCloudAPI(profile='dev01')
>>> from cbc_sdk.enterprise_edr import Feed, Watchlist
>>> feed = api.select(Feed, 'zcA0XzYTXKprqKN3ZjDtw')
>>> watchlist = Watchlist.create_from_feed(feed)
>>> watchlist._info
{'name': 'Feed Suspicious Applications2', 'description': 'Any signs of suspicious applications running on our endpoints', 'tags_enabled': True, 'alerts_enabled': False, 'classifier': {'key': 'feed_id', 'value': 'zcA0XzYTXKprqKN3ZjDtw'}}
>>> watchlist.save()
<cbc_sdk.enterprise_edr.threat_intelligence.Watchlist: id Egg10PLQQJW54Mzs3cjt2Q> @ https://defense-dev01.cbdtest.io
>>> watchlist._info
{'name': 'Feed Suspicious Applications2', 'description': 'Any signs of suspicious applications running on our endpoints', 'tags_enabled': True, 'alerts_enabled': False, 'classifier': {'key': 'feed_id', 'value': 'zcA0XzYTXKprqKN3ZjDtw'}, 'id': 'Egg10PLQQJW54Mzs3cjt2Q', 'create_timestamp': 1628201980, 'last_update_timestamp': 1628201980, 'report_ids': None}
```